lua/lazyvim/plugins/extras/lang/java.lua

```diff
@@ -1,125 +1,143 @@
-            pattern = filetypes,
-            callback = function()
-              local fname = vim.api.nvim_buf_get_name(0)
-              local root_dir = find_java_project_root(fname)
-              local project_name = root_dir and vim.fs.basename(root_dir)
-              local cmd = { "jdtls" }
-              if project_name then
-                local jdtls_cache_dir = vim.fn.stdpath("cache") .. "/jdtls/" .. project_name
-                vim.list_extend(cmd, {
-                  "-configuration",
-                  jdtls_cache_dir .. "/config",
-                  "-data",
-                  jdtls_cache_dir .. "/workspace",
-                })
-              end
-              local jdtls_base_config = {
-                on_attach = require("lazyvim.util").on_attach(function(client, buffer)
-                  if mason_registry.has_package("java-test") then
-                    -- custom keymaps for Java test runner (not yet compatible with neotest)
-                    vim.keymap.set("n", "<leader>tT", function() require("jdtls").pick_test({ bufnr = buffer }) end, { buffer = buffer, desc = "Run specific Test" })
-                    vim.keymap.set("n", "<leader>tt", function() require("jdtls").test_class({ bufnr = buffer }) end, { buffer = buffer, desc = "Run File" })
-                    vim.keymap.set("n", "<leader>tr", function() require("jdtls").test_nearest_method({ bufnr = buffer }) end, { buffer = buffer, desc = "Run nearest" })
-                  end
-                  if mason_registry.has_package("java-debug-adapter") then
-                    -- custom init for Java debugger
-                    require("jdtls").setup_dap({ hotcodereplace = "auto" })
-                    require("jdtls.dap").setup_dap_main_class_configs()
-                  end
-                  require("jdtls.setup").add_commands()
-                end),
-                cmd = cmd,
-                root_dir = root_dir,
-                init_options = {
-                  bundles = bundles,
-                }
-              }
-              local jdtls_opts = require("lazyvim.util").opts("nvim-jdtls")
-              require("jdtls").start_or_attach(vim.tbl_deep_extend("force", jdtls_opts or {}, jdtls_base_config))
-              require("which-key").register({ c = { x = { name = "Extract" } } }, { prefix = "<leader>" })
-            end,
-          })
-          return true -- avoid duplicate servers
+          return true
         end,
       },
     },
   },
+  {
+    "mfussenegger/nvim-jdtls",
+    dependencies = {
+      "neovim/nvim-lspconfig",
+      "mfussenegger/nvim-dap",
+      "hrsh7th/nvim-cmp",
+      "folke/which-key.nvim",
+    },
+    ft = { "java" },
+    config = function()
+      local project_name = vim.fn.fnamemodify(vim.fn.getcwd(), ":p:h:t")
+      local workspace_dir = vim.fn.stdpath("data") .. "/lsp/java/" .. project_name
+      local config = {
+        cmd = {
```

Comment:
providing the full cmd is usually no longer required as most of the default parameters are given by the jdtls.sh script now, so can you evaluate/explain if there is something in addition to those default parameters needed? I would rather rely on the jdtls bundled startup script instead of override everything…

Reply:
Hi, plz ref this https://github.com/LazyVim/LazyVim/pull/1300#discussion_r1286696050